src/UI/FrontEnd.java

```diff
@@ -472,6 +493,21 @@ private void initializeCreditCardAddTab(JTabbedPane creditCardTabbedPane) {
 		creditCardAddTab.add(creditCardAddZipInput);
 
 		JButton creditCardAddButton = new JButton("Add");
+		creditCardAddButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if(creditCardAddNumberInput.getText().isEmpty() || creditCardAddExpDateInput.getText().isEmpty() || creditCardAddCVVInput.getText().isEmpty() || creditCardAddStreetAdressInput.getText().isEmpty() || creditCardAddCityInput.getText().isEmpty() || creditCardAddStateInput.getText().isEmpty() || creditCardAddZipInput.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(frame, "All fields marked with * must have a value");
+				}else {
+					if(UIController.addCreditCard(creditCardAddNumberInput.getText(),creditCardAddNicknameInput.getText(),creditCardAddExpDateInput.getText(),creditCardAddCVVInput.getText(),creditCardAddStreetAdressInput.getText(),creditCardAddCityInput.getText(),creditCardAddStateInput.getText(),creditCardAddZipInput.getText())) {
```

Comment:
did you check that having no input correctly adds a card w default nickname?

Reply:
I'll do that real quick

Reply:
it works